Automattic/pocket-casts-ios
# SSO: Adds the social button style to the login view

| 📘 Project: #381 |
|:---:|

## Description
This adds the social login button styles from the designs, and updates the view layout to be more inline with the designs.

I also added the ability to use dynamically sizing fonts without breaking the layout, and updated the view to use dynamic fonts.

## Screenshots

### Flag enabled
| iPhone SE | iPhone 14 Pro |
|:---:|:---:|
|<img height="320" alt="Screen Shot 2022-11-02 at 8 44 45 PM" src="https://user-images.githubusercontent.com/793774/199628232-204c904a-aa84-442e-bfcc-b27980546f1d.png">|<img height="320" alt="Screen Shot 2022-11-02 at 8 44 35 PM" src="https://user-images.githubusercontent.com/793774/199628239-0e0b9ee9-d2a4-4b66-a4d4-d8ffff22abca.png">|

### Flag Disabled
| iPhone SE | iPhone 14 Pro |
|:---:|:---:|
|<img height="320" alt="Screen Shot 2022-11-02 at 8 44 45 PM" src="https://user-images.githubusercontent.com/793774/199628450-9789ea92-a1c3-4153-8163-6039484758f6.png">|<img height="320" alt="Screen Shot 2022-11-02 at 8 44 35 PM" src="https://user-images.githubusercontent.com/793774/199628333-4489cc2d-20d3-46a1-965e-c8d2337ca610.png">|

### Accessibility Largest Text + Bold Text

| iPhone SE | iPhone 14 Pro |
|:---:|:---:|
|<img height="320" alt="Screen Shot 2022-11-02 at 8 44 45 PM" src="https://user-images.githubusercontent.com/793774/199628874-4ca27005-b9a0-4b47-b6a3-8f2827db6b4e.png">|<img height="320" alt="Screen Shot 2022-11-02 at 8 44 35 PM" src="https://user-images.githubusercontent.com/793774/199628920-caa33d14-3c60-4f99-bf60-24729e2973e6.png">|

## To test

1. Launch the app with the `signInWithApple` FeatureFlag disabled
2. Sign out if you're signed in
3. Tap the login
4. ✅ Verify the view looks good
5. Tap the Create account and sign in buttons
6. ✅ Verify they work still
7. Change your accessibility text size (you should close and reopen the login landing view)
8. ✅ Verify the view looks good and the layout doesn't break
9. Enable the signInWithApple FeatureFlag
10. Repeat the steps from above
11. ✅ Verify the view looks good
12. Repeat the steps using different themes to verify the social login buttons react accordingly

## Checklist
